ZuluContent/Spells/Third/Teleport.cs

```diff
@@ -8,120 +9,82 @@
 
 namespace Server.Spells.Third
 {
-    public class TeleportSpell : MagerySpell
+    public class TeleportSpell : MagerySpell, ITargetableAsyncSpell<IPoint3D>
     {
-        public TeleportSpell(Mobile caster, Item spellItem) : base(caster, spellItem)
+        public TeleportSpell(Mobile caster, Item spellItem) : base(caster, spellItem) { }
+        
+        public async Task OnTargetAsync(ITargetResponse<IPoint3D> response)
         {
-        }
+            if (!response.HasValue)
+                return;
 
-        public override bool CanCast()
-        {
-            if (!base.CanCast())
-                return false;
+            var point = response.Target;
             
-            if (WeightOverloading.IsOverloaded(Caster))
-            {
-                Caster.SendLocalizedMessage(502359, "", 0x22); // Thou art too encumbered to move.
-                return false;
-            }
-
-            return SpellHelper.CheckTravel(Caster, TravelCheckType.TeleportFrom);
-        }
-
-        public override void OnCast()
-        {
-            Caster.Target = new InternalTarget(this);
-        }
-
-        public void Target(IPoint3D p)
-        {
-            var orig = p;
+            var orig = point;
             var map = Caster.Map;
 
-            SpellHelper.GetSurfaceTop(ref p);
+            SpellHelper.GetSurfaceTop(ref point);
 
             var from = Caster.Location;
-            var to = new Point3D(p);
+            var to = new Point3D(point);
 
             if (WeightOverloading.IsOverloaded(Caster))
```

Comment:
Ah interesting - did this exist in ZH? I used to think it would be cool to teleport when u can't move from stam loss :P

Reply:
Ah yeah good spot I just brought that over from the OSI code. Yeah I used to tele around overloaded in my house :joy: